compiler/src/dotty/tools/dotc/typer/ErrorReporting.scala

```diff
@@ -28,11 +28,14 @@ object ErrorReporting {
 
   def cyclicErrorMsg(ex: CyclicReference)(implicit ctx: Context) = {
     val cycleSym = ex.denot.symbol
-    def errorMsg(msg: String, cx: Context): Message =
+    def errorMsg(msg: Message, cx: Context): Message =
       if (cx.mode is Mode.InferringReturnType) {
         cx.tree match {
           case tree: untpd.DefDef if !tree.tpt.typeOpt.exists =>
-            OverloadedOrRecursiveMethodNeedsResultType(tree.name)
+            // TODO: analysis if tree is an overloaded method (or directly recursive)
```

Comment:
It complains that we assign an error type into a tree, yet not errors were reported. Is that stacktrace reached from the call
 
    cycleSym.owner.info.member(cycleSym.name).isOverloaded

? If yes, can you show me the full stacktrace up to the call?